conventions/cli.md

```diff
@@ -1,79 +1,181 @@
+This first noun forms a command group for all the operations you might want to do with that kind
+of resource.
+Sometimes there can be deeper hierarchies with multiple nouns (`kn <noun-1> <noun-2> .... <verb>`) when it makes sense to structure complex concepts.
+A good example is `kn source <source-type> <verb>` which is used like in `kn source ping create`.
 
-The <group>.knative.dev Kind, singluar and lowercase. For example, `service` for
-`serving.knative.dev/service` or `trigger` for `eventing.knative.dev/trigger`.
+`kn` commands take only positional arguments that are used as [identifiers](#identifier).
+This identifier is often the name of a resource which identifies the resource uniquely for the current or given namespace.
 
-## Verb
+Top-level commands concerning the operation of `kn` itself, like `help` and `version` are also okay.
 
-If the thing the user's doing can be described by the following commands, these
-should be the name of the verb:
+### Noun
 
-- `describe` prints detailed information about a single resource. It can include
-  status information of related or child resources, too.
+For resource-related commands, the kind itself used as a command in singular and lowercase form. For example, `service` for
+`serving.knative.dev/service` or `trigger` for `eventing.knative.dev/trigger` are the commands for managing these resources respectively
 
-- `list` prints summary information about all resources of a type, possibly
-  filtered by parent or label selector.
+### Verb
 
-- `create` creates a resource. Accepts a `--force` flag to create-or-replace.
+For CRUD (create-retrieve-update-delete) operation the following verbs have to be used:
 
-- `update` updates a resource based on the changes the user would like to make.
+- `describe` prints detailed information about a single resource that can contain data of dependent objects, too.
+- `list` prints summary information about all resources of a type.
+- `create` creates a resource.
+- `update` updates a resource.
+- `delete` deletes a resource,
 
-- `delete` deletes a resource
+For a given resource, create and update should use the same arguments as much as possible and where it makes sense.
 
-For a given resource there should be parallelism between arguments to `create`
-and `update` as much as possible.
+Other domain-specific verbs are possible on a case-by-case basis for operations that go beyond basic CRUD operations.
 
-Other domain-specific verbs are possible on a case-by-case basis, like
-`set-traffic` for a Knative Serving Service.
+### Identifier
 
-## Arguments
+For the `CRUD` operations `describe`, `create`, `update`, `delete` the identifier is the resource's name and is required as a positional argument after the commands.
+For example it is the last argument that does not start with a flag prefix `-` or `--`.
+`list` operations don't use a resource name.
```

Comment:
>For the `CRUD` operations `describe`, `create`, `update`, `delete` the identifier is the resource's name and is required as a positional argument after the commands.
For example it is the last argument that does not start with a flag prefix `-` or `--`.
`list` operations don't use a resource name.

`list` operations can use the resource name to filter the resource among list (`kn service list foo`)